### PR TITLE
[cmd/mdatagen] Mdatagen support histogram metrics

### PR DESCRIPTION
**Pull request by sakulali**
**Description:** 
Metadata generator support histogram metrics

**Link to tracking Issue:**
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/26118

**Testing:** 
make generate
make mdatagen-test
make chlog-validate

**Documentation:**
cmd/mdatagen/documentation.md

### CODE REVIEW

File: cmd/mdatagen/templates/metrics.go.tmpl
Comment (dmitryax) [inline cmd/mdatagen/templates/metrics.go.tmpl:79]: What about bucket values? Histograms without buckets are invalid

**Review by atoulme — APPROVED** [review]
Changes look good to me, @dmitryax should review as codeowner.

**Pull request closed without merging**
